deepchem/data/data_loader.py

```diff
@@ -908,11 +987,54 @@ def create_dataset(self,
       input_files = [input_files]
 
     def shard_generator():
+      sequences = np.array([])
       for input_file in input_files:
-        X = encode_bio_sequence(input_file)
-        ids = np.ones(len(X))
-        # (X, y, w, ids)
-        yield X, None, None, ids
+        sequences = np.append(sequences, _read_file(input_file))
+      X = self.featurizer(sequences)
+      ids = np.ones(len(X))
+      # (X, y, w, ids)
+      yield X, None, None, ids
+
+    def _read_file(input_file: str, auto_add_annotations: bool = False):
+      """
+      Convert the FASTA file to a numpy array of FASTA-format strings.
+      """
+
+      def _generate_sequences(fasta_file, header_mark=">") -> np.array:
+        """
+        Uses a fasta_file to create a numpy array of annotated FASTA-format strings
+        """
+        sequences = np.array([])
+        sequence: List[str] = []
+        header_read = False
+        for line in fasta_file:
+          # Check if line is a header
+          if line.startswith(header_mark):  # New header line
+            header_read = True
+            sequences = _add_sequence(sequences, sequence)
+            sequence = []
+          elif header_read:  # Line contains sequence in FASTA format
+            if line[-1:] == '\n':  # Check last character in string
+              line = line[0:-1]  # Remove last character
+            sequence.append(line)
+        sequences = _add_sequence(sequences, sequence)
+        return sequences
+
+      def _add_sequence(sequences: np.array, sequence: list) -> np.array:
+        # Handle empty sequence
+        if sequence is None or len(sequence) <= 0:
+          logger.warning(
```

Comment:
Could this spam many warnings? We want to make sure user logs won't be flooded with excess warnings if we expect this case to happen frequently

Reply:
For now, I changed it into logger.info.

I think it's generally quite unexpected for the FASTA file to be filled with empty sequences.